src/NServiceBus.Core/Unicast/ContextualBus.cs

```diff
@@ -101,43 +102,48 @@ public Task ReplyAsync(object message, NServiceBus.ReplyOptions options)
         /// <typeparam name="T">The type of message, usually an interface.</typeparam>
         /// <param name="messageConstructor">An action which initializes properties of the message.</param>
         /// <param name="options">Options for this reply.</param>
-        public Task ReplyAsync<T>(Action<T> messageConstructor, NServiceBus.ReplyOptions options)
+        /// <param name="incomingContext">The context of the incoming message</param>
+        public Task ReplyAsync<T>(Action<T> messageConstructor, NServiceBus.ReplyOptions options, BehaviorContext incomingContext)
         {
-            return ReplyAsync(messageMapper.CreateInstance(messageConstructor), options);
+            return ReplyAsync(messageMapper.CreateInstance(messageConstructor), options, incomingContext);
         }
 
         /// <summary>
         /// Moves the message being handled to the back of the list of available 
         /// messages so it can be handled later.
         /// </summary>
-        public async Task HandleCurrentMessageLaterAsync()
+        public async Task HandleCurrentMessageLaterAsync(InvokeHandlerContext incomingContext)
```

Comment:
I wanted to extract these methods to a new class anyway as a next step. Let's rename it there?